### PR TITLE
Update egg-teamspeak3-server.json

### PR DESCRIPTION
**Pull request by xopez**
Allow to users to change filetransfer port, telnet port and ssh port.
I also put the telnet port to the default one and not the voice port.
I saw one problem which I don't know to fix.
For the change of the filestransfer port you need to give an ip. So I put 0.0.0.0
Normally Teamspeak binds `2019-03-22 11:47:57.590315|INFO |VirtualServer |1 |listening on 0.0.0.0:9988, [::]:9988`
But i didn't find a way to give the second one with it. 

Log that you can see it's is starting with custom ports.
```
[Pterodactyl Daemon] Server marked as STARTING
[Pterodactyl Daemon] Checking size of server data directory...
[Pterodactyl Daemon] Disk Usage: 18M / 0M
[Pterodactyl Daemon] Ensuring file permissions.
[Pterodactyl Daemon] Running server preflight.
[Pterodactyl Daemon] Starting server container.
2019-03-22 11:54:14.502446|INFO | | |myTeamSpeak identifier revocation list was downloaded successfully - all related features are activated
2019-03-22 11:54:14.968633|INFO | | |Puzzle precompute time: 521
2019-03-22 11:54:14.968926|INFO |FileManager | |listening on 0.0.0.0:30034
[Pterodactyl Daemon] Server marked as ON
2019-03-22 11:54:14.979194|INFO |VirtualServer |1 |listening on 0.0.0.0:9988, [::]:9988
2019-03-22 11:54:14.979486|INFO |Query | |listening for query on 0.0.0.0:10012, [::]:10012
2019-03-22 11:54:14.979806|INFO |Query | |listening for query ssh on 0.0.0.0:10023, [::]:10023
2019-03-22 11:54:14.979941|INFO |CIDRManager | |updated query_ip_whitelist ips: 127.0.0.1/32, ::1/128,
```

EDIT: I also updated to the latest teamspeak version

EDIT2: I try to change it so it runs with an inifile where you can configure it. So you don't need the startup parameters anymore.

### CODE REVIEW

**Comment by ZeProf2Code** [comment]
`https:\/\/files.teamspeak-services.com\/releases\/server\/${TS_VERSION}\/teamspeak3-server_linux_amd64-${TS_VERSION}.tar.bz2`
Use this as download URL instead of
`http:\/\/dl.4players.de\/ts\/releases\/${TS_VERSION}\/teamspeak3-server_linux_amd64-${TS_VERSION}.tar.bz2`

**Comment by xopez** [comment]
done

**Comment by ZeProf2Code** [comment]
👍

**Comment by xopez** [comment]
Well, I updated it a little bit. On install creates ts3server.ini with default config. Only creates it, if not there, so users don't need to worry about overwrite when updating the server. Also starts now the binary with the inifile.

**Comment by xopez** [comment]
I have finished the new updating way. For short explanation.

- Starts the binary with inifile directly
- `ts3server.ini` and `.ts3server_license_accepted` automatically generated if not existent. Just needed than so that for example the config doesn't get overwritten
- Version info can only get in the `logs/ts3server*_0.log`
- if log not there then install latest
- if there and update on then install newest
- if update off then get the current link and replace the version with the latest from the logfiles
- added more apk requirements
- changed container to `ubuntu:18.04` and script Script Entrypoint Command to `bash` cause I need the command `sort -V/sort --version-sort` which isn't avaiable in ash.

Tested all possible ways for my environment. 
For testing.
If you want to reinstall a old version:
Just install current and start/stop the server. Go to the log `ts3server_*0.log`.
Change the current version to 3.5.0 or something else. Turn update-parameter to `0` and reinstall. Start so can see the old version.
Important. For downgrade delete `ts3server.sqlitedb, ts3server.sqlitedb-shm, ts3server.sqlitedb-wal` cause you can't downgrade a teamspeak db. It gets updated with mostly any version.

**Comment by ZeProf2Code** [comment]
what?

**Comment by xopez** [comment]
I copied the wrong, changed it already.
My variables are empty. And I don't know why. It works on ubuntu itself quite fine.

**Comment by parkervcp** [comment]
@xopez put install logs into a gist or something please. It makes scrolling these difficult.

Instead of make it so a user could decide on what version to install you just defaulted to install the latest or keep an old version. It's a no from me for that alone.

Also never default to accepting a license for a user. It's also no from me for this as well.

File: database/seeds/eggs/voice-servers/egg-teamspeak3-server.json [hunk 1]
```diff
@@ -3,34 +3,34 @@
...
+            "entrypoint": "bash"
         }
     },
     "variables": [
         {
-            "name": "Server Version",
-            "description": "The version of Teamspeak 3 to use when running the server.",
-            "env_variable": "TS_VERSION",
-            "default_value": "3.5.0",
+            "name": "Update",
```
Comment (parkervcp) [inline database/seeds/eggs/voice-servers/egg-teamspeak3-server.json]: I commented on defaulting to latest but letting the user choose and you did this instead.

File: database/seeds/eggs/voice-servers/egg-teamspeak3-server.json [hunk 2]
```diff
@@ -3,34 +3,34 @@
...
...e \"\\nDownloading TS3 Version $INSTALLEDVERSION\\n\"\r\n\t\t\tcurl $OLDTS3RELEASELINK | tar xj --strip-components=1\r\n\t\tfi\r\n\tfi\r\n\t# Just do it one time\r\n\tbreak\r\ndone\r\n\r\n# Accept license\r\nif [ ! -f \".ts3server_license_accepted\" ]; then\r\n\ttouch \".ts3server_license_accepted\"\r\nfi\r\n\r\n# Add default config if not there\r\nif [ ! -f \"ts3server.ini\" ]; then\r\n    echo \"Creating default config for teamspeak\"\r\n    echo 'machine_id=\r\ndefault_voice_port=9987\r\nvoice_ip=\r\nlicensepath=\r\nfiletransfer_port=30033\r\nfiletransfer_ip=\r\nquery_port=10011\r\nquery_ip=0.0.0.0, ::\r\nquery_ip_whitelist=query_ip_whitelist.txt\r\nquery_ip_blacklist=query_ip_blacklist.txt\r\ndbplugin=ts3db_sqlite3\r\ndbpluginparameter=\r\ndbsqlpath=sql\/\r\ndbsqlcreatepath=create_sqlite\/\r\ndbconnections=10\r\nlogpath=logs\r\nlogquerycommands=0\r\ndbclientkeepdays=30\r\nlogappend=0\r\nquery_skipbruteforcecheck=0\r\nquery_buffer_mb=20\r\nhttp_proxy=\r\nlicense_accepted=1\r\nserverquerydocs_path=serverquerydocs\/\r\nquery_ssh_ip=0.0.0.0, ::\r\nquery_ssh_port=10022\r\nquery_protocols=raw,ssh\r\nquery_ssh_rsa_host_key=ssh_host_rsa_key\r\nquery_timeout=300' > ts3server.ini\r\nfi",
```
Comment (parkervcp) [inline database/seeds/eggs/voice-servers/egg-teamspeak3-server.json]: You default to accepting a license for a user. That would make anyone here liable for their user who never had to accept the license.

Comment (xopez) [reply database/seeds/eggs/voice-servers/egg-teamspeak3-server.json]: Before the license was accepted via command line:
https://github.com/pterodactyl/panel/blob/develop/database/seeds/eggs/voice-servers/egg-teamspeak3-server.json#L11

File: database/seeds/eggs/voice-servers/egg-teamspeak3-server.json [hunk 3]
```diff
@@ -27,10 +27,37 @@
             "name": "Server Version",
             "description": "The version of Teamspeak 3 to use when running the server.",
             "env_variable": "TS_VERSION",
-            "default_value": "3.5.0",
+            "default_value": "",
```
[the remark this answers is not in the document]
Comment (xopez) [reply database/seeds/eggs/voice-servers/egg-teamspeak3-server.json]: Am I understanding you right?
You want that you user can input nothing and latest for latest version or a version to get this specific version?

**Comment by DaneEveritt** [comment]
Please re-open this against `pterodactyl:0.7-develop` which has all of the old 0.7.x code, as well as a bunch of TS changes to the egg that probably need to be worked around with these changes anyways.

**Pull request closed without merging**